lib/AST/AlignTokens.cpp

```diff
@@ -1144,6 +1064,22 @@ bool Matcher::MatchBalanced(BalancedRegion *parsed, BalancedRegion *printed,
     }
   }
 
+  // Force match the parens/brackets/braces. Here we're targeting the attribute
+  // case, where we might have multiple attributes syntax blocks printed, but
+  // only one attribute syntax block in the parsed code.
+  if (has_begin_loc && begin_loc_matches && !has_end_loc) {
+    MergeToken(parsed->end, printed->end, changed, true  /* force */);
+    end_loc_matches = true;
+
+  } else if (has_end_loc && end_loc_matches && !has_begin_loc) {
+    MergeToken(parsed->begin, printed->begin, changed, true  /* force */);
+    begin_loc_matches = true;
+  }
```

Comment:
C&P.

Reply:
All the variables are different; I think generalizing won't reduce code complexity.